### DISCUSSION TITLE
How to yield output of of 1st query in 2nd query?

### DISCUSSION THREAD

**Discussion by akhilravuri-tul-scm** [vesoft-inc/nebula#4364] [Q&A] [answered]
Hi,

```go from "AdminID" over AdminToUserEdge yield $$.user.id as userId, "destPlace" as v3, properties($$) as userDetails | FETCH PROP ON userDestPlaceEdge $-.userId -> $-.v3 @0 YIELD properties(edge) as edgeprop, $-.userDetails as userDetails ```

Output:- ```SemanticError: unsupported input/variable property expression in yield.```

Is there any way to return the userDetails from Fetch query?

Thanks,
Akhil

**Comment by xiajingchun** [comment, marked as answer]
Looks like the same request as described in #3984 , not supported yet.
Maybe you can use match to rewrite your query, although usally there will some performance loss comparing to GO & FETCH.

**Comment by akhilravuri-tul-scm** [comment]
@xiajingchun  How many properties can we create in a single edge?

If we use fetch on edge by giving scrId and dstId will it effect performance as properties increases?


Thanks,
Akhil

**Reply by wey-gu** [reply]
> @xiajingchun How many properties can we create in a single edge?
> 
> If we use fetch on edge by giving scrId and dstId will it effect performance as properties increases?
> 
> Thanks, Akhil

Fetching edge properties underlying is a key seek in RocksDB, the srcid, dstid, edge_type, rank will be the key of the record and property in blob will be the value.

When the property is large(that is we define a huge amount of property columns per edge type), the performance in pure FETCH could be impacted from the caching perspective, that is, fewer data could be placed in memory. To mitigate that, `Key value separation` could be leveraged.

ref: https://docs.nebula-graph.io/3.2.0/5.configurations-and-logs/1.configurations/4.storage-config/#key-value_separation_configurations

note, we should at best reduce property when possible during the processing of data before putting them into graph databases, but when it has to be put into properties fields(with a bunch of them, try leveraging k-v separation :).

**Comment by akhilravuri-tul-scm** [comment]
@wey-gu Yes we can use key-value separation but that leads us to 2billion edges and this might increase when we onboard a new partner. So we wanted to add the details of new partner to the edge by adding properties.

Is there any hard limit for no.of properties in a edge. I know you said internally it stores it in form of key-value. But is there any limit for that value size? (Like Couchbase has limit that doc size cannot be more than 20MB).

It's okay if our data is not stored in the cache and every time it is fetched from storage.

Where do we find more info about cache mechanism used by nebula graph?

Thanks,
Akhil

**Reply by wey-gu** [reply]
Now NebulaGraph leveraged RocksDB underlying only, the limitation discussion could be found here: https://github.com/facebook/rocksdb/issues/513 
https://github.com/facebook/rocksdb/wiki/RocksDB-FAQ

> **Q: What's the maximum key and value sizes supported?**
> 
> A: In general, RocksDB is not designed for large keys. The maximum recommended sizes for key and value are 8MB and 3GB respectively.

Regarding the cache mechanism now, in the storage layer, there are
- block cache, ref: https://github.com/facebook/rocksdb/wiki/Block-Cache
- vertex cache (cached key vid, value properties, by default disabled)
- empty cache (cached vid + tag, not existed)

I will upload our community meetup video later, in the last meeting, the topic was about the cache :).

**Reply by akhilravuri-tul-scm** [reply]
Looking forward to watch the video.

**Reply by wey-gu** [reply]
> Looking forward to watch the video.

Will upload soon, and, I just realized after replying you this, that the other two cache are enterprise-version only features😅

**Reply by akhilravuri-tul-scm** [reply]
@wey-gu sorry for tagging you.

Does nebula support Triggers?

We need to update ed1 edge when there are updates on ed2 edge. Does nebula support this?

**Reply by wey-gu** [reply]
NP, there isn't semantic support on this now, did you see anything similar on another database as a reference that maybe could raise an issue for a feature request?

This has to be done via application level together with the ed1 change hook now i think.

**Reply by wey-gu** [reply]
> Looking forward to watch the video.

Here you go, https://youtu.be/-DAY5sQ_YOc :)

**Comment by akhilravuri-tul-scm** [comment]
Hi @wey-gu,

we have deployed 3 stroraged, 3 graphed, 1 metad node server in our cluster.
We created a space with 2 partitions and 3 replicas.

when we try to delete storage pod one by one after the deleted one is ready we get the data till 2 deletions. But when delete the last pod(3rd one) we get leader changed error while querying data.

can you please let us know why? nebula support high availability if we replicas and 3 node cluster right?

Thanks,
Akhil
